dfastmi/io/ApplicationSettingsHelper.py

```diff
@@ -0,0 +1,159 @@
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+from typing import Any, Dict, List, Optional, TextIO
+
+class ApplicationSettingsHelper:
+    """
+        Load texts from configuration file, and store globally for access.
+    """
+    PROGTEXTS: Dict[str, List[str]]
+
+    @staticmethod
+    def load_program_texts(filename: str) -> None:
+        """
+        Load texts from configuration file, and store globally for access.
+
+        This routine reads the text file "filename", and detects the keywords
+        indicated by lines starting with [ and ending with ]. The content is
+        placed in a global dictionary PROGTEXTS which may be queried using the
+        routine "get_text". These routines are used to implement multi-
+        language support.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the file to be read and parsed.
+        """
+        text: List[str]
+        dict: Dict[str, List[str]]
+
+        all_lines = open(filename, "r").read().splitlines()
+        dict = {}
+        text = []
+        key = None
+        for line in all_lines:
+            rline = line.strip()
+            if rline.startswith("[") and rline.endswith("]"):
+                if not key is None:
+                    dict[key] = text
+                key = rline[1:-1]
+                text = []
+            else:
+                text.append(line)
+        if key in dict.keys():
+            raise Exception('Duplicate entry for "{}" in "{}".'.format(key, filename))
+        if not key is None:
+            dict[key] = text
+        ApplicationSettingsHelper.PROGTEXTS = dict
+
+
+    @staticmethod
+    def get_text(key: str) -> List[str]:
+        """
+        Query the global dictionary of texts via a string key.
+
+        Query the global dictionary PROGTEXTS by means of a string key and return
+        the list of strings contained in the dictionary. If the dictionary doesn't
+        include the key, a default string is returned.
+
+        Parameters
+        ----------
+        key : str
+            The key string used to query the dictionary.
+
+        Returns
+        -------
+        text : List[str]
+            The list of strings returned contain the text stored in the dictionary
+            for the key. If the key isn't available in the dictionary, the routine
+            returns the default string "No message found for <key>"
+        """
+
+        try:
+            application_setting = ApplicationSettingsHelper.PROGTEXTS[key]
+        except:
+            application_setting = ["No message found for " + key]
+        return application_setting
```

Comment:
You could change this to the one liner ApplicationSettingsHelper.PROGTEXTS.get(key, "No message found for " + key)

This a more safe way to request data from a dict in python, this way you prevent a KeyError if key does not exist.
The second parameter in this method is returned if the key does not exist.
this also makes the try except case here redundant.

https://www.w3schools.com/python/ref_dictionary_get.asp
https://stackoverflow.com/questions/11041405/why-dict-getkey-instead-of-dictkey

Reply:
geniaal

Reply:
Refactoring & improvements should be done in separate issue for each component